Cleaning up css

f8daniel opened the pull request:
#253 done

I have changed the p1, p2, ... class names used to position players around a virtual table because they were conflicting with settlers and imperium using p1, p2... for color coding, so I updated all the card games (poker, hearts, president + blackjack via game-player-box) to use poker-seat-1, poker-seat-2... and updated the css selectors in game.css. The layout and gameplay of Imperium and Poker are (or should be) completely unaffected by the changes. 

Furthermore, it seems @trevelyan and I have been going back in forth in the css for positioning game-player-boxes (several rounds now), but I think this should sort things out. Even though I removed the id-based css selectors for playerboxes in game.css (which was messing up the wuziqi and settlers apps), gametestsuite still demonstrates playerboxes nicely.

escapedcat commented on lib/saito/ui/game-playerbox/game-playerbox.js:
```diff
@@ -7,8 +7,44 @@
 
-    Use render to create the playerboxes
-    Use attachEvents to make them movable
+  API:
```
I might confuse this, but I assume if you stick to the [JSDoc](https://jsdoc.app/) format this woudl be auto-generated to here: https://saitotech.github.io/saito-lite/index.html  

This would make it more visible if other devs look at the created docs.

escapedcat commented on lib/saito/ui/game-playerbox/game-playerbox.js:
```diff
@@ -51,14 +87,32 @@ class GamePlayerBox {
             document.getElementById(("player-box-"+player)).style.display = "none";
-	  }
-	}
+	        }
```
If you throw this into [prettier](https://prettier.io/) the format could be more standardized.

escapedcat commented on lib/saito/ui/game-playerbox/game-playerbox.js:
```diff
@@ -51,14 +87,32 @@ class GamePlayerBox {
+      try {
+        for (let i = 1; i <= this.game_mod.game.players.length; i++) {
+          let player = this._playerBox(i);
+          if (document.getElementById(("player-box-"+player))) {
```
Could use a template-literal, i.e.:
```js
`player-box-${player}`
```

escapedcat commented on mods/blackjack/web/style.css:
```diff
@@ -124,7 +124,7 @@ body {
     width: 8vw;
```
Would be nice to throw this fiel into prettier as well. And other files below as well.

trevelyan commented:
Just a random thought from seeing this suggestion:

-    Use render to create the playerboxes
-    Use attachEvents to make them movable

My personal belief is that playerboxes are not intended to be
draggable by default. Their original conceptualization is just as
player-by-player information boxes that get positioned in various
parts of the screen to assist in particular with card games.

Games that need to make components draggable can do so:

 app.browser.makeDraggable(div_id)

What tends to happen naturally with development is that components get
extended -- what seems to be happening is the conversion of the P1
playerbox into a lite-version-HUD. There's nothing wrong with this at
all -- it's a positive development because it works better for some
games with limited kinds of interactivity, but that doesn't mean we
want to make things like that default behavior.

displayPlayerHand() {
  this.playerbox.render(this.app, this);
  this.playerbox.attachEvents(this.app, this);
  // make draggable here
}

The most difficult component to maintain is currently the
game-cardbox, because it is hooked up to handle card-popup on
mouseover when people put cards into the HUD, and has custom behavior
for both desktop and mobile -- mobile users need a sanity check to
confirm they really meant to click the card, since they need that
first click to show it. The more complicated and specific the
components get. Even now it's a bit of a nightmare even implementing
the cardbox, although it has gotten easier with the Web3 work.

f8daniel commented:
- Okay, should have the documentation formatted correctly for JSDoc now.
- Using more template literals.
- Removed draggability from attachEvents (because not consistent with function of same name in other UI tools), but put it in new function. 
- Fixed a bug with the player-numbering given a user defined seating chart.
- Fixed a bug with player numbers in wuziqi

trevelyan commented:
> I have changed the p1, p2, ... class names used to position players around a virtual table because they were conflicting with settlers and imperium using p1, p2... for color coding, so I updated all the card games (poker, hearts, president + blackjack via game-player-box) to use poker-seat-1, poker-seat-2... and updated the css selectors in game.css. 

Yeah. Will also introduce you to Duke and Tucho etc. For changes like this we can ask them to help UI test all of the games.

I've just deployed onto the server, so will merge this. If there are any issues we should be able to catch them before the next push.

Thanks Dan.
